app/models/manufacturer.rb

```diff
@@ -26,6 +26,18 @@ def volume
     donations.joins(:line_items).sum(:quantity)
   end
 
+  def self.by_donation_date(count = 10, date_range = nil)
+    # selects manufacturers that have donation qty > 0 in the provided date range
+    # and sorts them by the date of the most recent donation
+    joins(donations: :line_items).where(donations: { issued_at: date_range })
+      .select('manufacturers.*, sum(line_items.quantity) as donation_count')
+      .select('manufacturers.*, max(donations.issued_at) as donation_date')
+      .group('manufacturers.id')
+      .having('sum(line_items.quantity) > 0')
+      .order('donation_date DESC')
+      .limit(count)
+  end
+
   def self.by_donation_count(count = 10, date_range = nil)
```

Comment:
If we are no longer using this `by_donation_count` method, I would delete it as it would be dead code.

Reply:
I am still hesitant to delete it given it expresses a distinct idea we might come back to, though can certainly see the argument for just cleaning it out.

Reply:
Let's delete it. If there's no current use for a function, it shouldn't be in the codebase.